### PR TITLE
scrolling on pdf pages

### LINKED ISSUE #954: Scrolling (etc) doesn't work on a PDF page
For instance, load [this](https://www.cs.rochester.edu/research/cisd/pubs/2007/allen-et-al-plow-aaai2007.pdf) and try to scroll down. You'll get a missing host permission.

I think if we add a permission for `"resource://pdf.js/"` to manifest.json.ejs that it will fix this.

### PR DESCRIPTION
**Pull request by danielamormocea**
Fixes #954 

This adds logic to distinguish between pdf pages and the others.
Don't want to interfere with the already assigned person.

### CODE REVIEW

File: extension/intents/scroll/scrollHelper.js [hunk 1]
```diff
@@ -3,7 +3,13 @@
 const scrollAmount = 0.9;
 
 function scrollVertically(dy, smooth) {
-  window.scrollBy({
+  let toBeScrolled = window;
+
+  if (document.contentType === "application/pdf") {
```
Comment (GangaChatrvedi) [inline extension/intents/scroll/scrollHelper.js]: Really elegant way to solve the problem.  

My only thought is. There can be many other contents as well that can be covered by your code (in addition to pdf). What if we put add html related contentType with **window** and ```toBeScrolled = document.activeElement``` for remaining.

**Comment by danielamormocea** [comment]
@GangaChatrvedi  made the requested change, if you can take a look :)

**Comment by danielamormocea** [comment]
@ianb @GangaChatrvedi  Could you please take a look at the changes above?

I forgot initially to make this work for scroll top and scroll bottom; I saw it later on the other PR.

File: extension/intents/scroll/scrollHelper.js [hunk 2]
```diff
@@ -2,8 +2,28 @@
...
+    scrollPos = element.scrollY;
+  }
+
+  return {
+    element,
+    height,
+    scrollPos,
+    scrollHeight,
+  };
+}
```
Comment (ianb) [inline extension/intents/scroll/scrollHelper.js]: While this works for the PDF example, it won't work for other pages, like a Bugzilla bug (https://bugzilla.mozilla.org/show_bug.cgi?id=343205) or a Google doc. I think the `document.contentType` check isn't the right thing: rather `document.activeElement` _or one of its ancestors_ is always the right thing to scroll.

This comment also gives some guidance: https://github.com/mozilla/firefox-voice/issues/954#issuecomment-591691095

**Comment by ianb** [comment]
#1235 is also related to this

**Comment by danielamormocea** [comment]
@ianb  I tried making this work for google docs, but I couldn't. It works for the rest of the sites well, but google has a weird object, and I can't figure it out :(

**Comment by ianb** [comment]
Oh yeah, I just looked at Google Docs, it is very weird. Full of weird invisible iframes... I wonder if they use them to capture different input and then just create their own rendering system on top of that.

But yeah, looks good. Approved but holding for #1245

**Pull request merged**